File: data/EXP01/golang_no_system_var/2466/2466.output_3.go
func countGoodStrings(low int, high int, zero int, one int) int {
    const mod = int(1e9 + 7)
    n := high - low + 1
    dp := make([][][]int, n+1)

    for i := range dp {
        dp[i] = make([][]int, zero+1)
        for j := range dp[i] {
            dp[i][j] = make([]int, one+1)
        }
    }

    dp[0][0][0] = 1

    for i := 1; i <= n; i++ {
        for z := 0; z <= zero; z++ {
            for o := 0; o <= one; o++ {
                dp[i][z][o] = (dp[i][z][o] + dp[i-1][z][o]) % mod

                if o > 0 && i <= high-low+1-z {
                    dp[i][z][o] = (dp[i][z][o] + dp[i-1][z][o-1]) % mod
                }
            }
        }
    }

    count := 0
    for z := 0; z <= zero; z++ {
        for o := 0; o <= one; o++ {
            if low <= z+o && z+o <= high {
                count = (count + dp[n][z][o]) % mod
            }
        }
    }

    return count
}

File: data/EXP01/python/2466/2466.output_3.py
# @lc app=leetcode id=2466 lang=python3
class Solution:
    def countGoodStrings(self, low: int, high: int, zero: int, one: int) -> int:
        mod = 10**9 + 7
        dp = [[[0 for _ in range(2)] for _ in range(one + 1)] for _ in range(zero + 1)]
        dp[0][0][0] = 1

        for i in range(zero + 1):
            for j in range(one + 1):
                for k in range(2):
                    if i > 0:
                        dp[i][j][0] = (dp[i][j][0] + dp[i-1][j][0]) % mod
                    if j > 0:
                        dp[i][j][1] = (dp[i][j][1] + dp[i][j-1][1]) % mod
                    if k == 0:
                        dp[i][j][0] = (dp[i][j][0] + dp[i][j][1]) % mod
                    elif k == 1:
                        dp[i][j][1] = (dp[i][j][1] + dp[i][j][0]) % mod

        ans = 0
        for i in range(low, high + 1):
            ans = (ans + dp[zero][one][0] + dp[zero][one][1]) % mod

        return ans